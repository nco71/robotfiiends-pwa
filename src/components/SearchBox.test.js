import React from 'react';
import {shallow} from 'enzyme';
import SearchBox from './SearchBox';

describe('<SearchBox />', () => {

    const mockSearchChange = jest.fn();

    const wrapper = shallow(<SearchBox searchChange={mockSearchChange} />);

    it('get the right event', ()=> {
        const event = {
            target: { value: 'the-value' }
          };
        wrapper.find('[id="search-robots-button"]').simulate('change', event);
       expect(mockSearchChange).toBeCalledWith(event)
    });

    it('render a component', () => {
        expect(wrapper.length).toBe(1);
    });


    it('renders component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    
    
    
  //test value of change , make mockup functions.

    })
