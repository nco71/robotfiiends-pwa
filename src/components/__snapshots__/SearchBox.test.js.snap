// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    searchChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        aria-label="Search"
        className="pa3 ba b--green bg-lightest-blue"
        id="search-robots-button"
        onChange={[Function]}
        placeholder="search robots"
        type="search"
/>,
      "className": "pa2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Search",
        "className": "pa3 ba b--green bg-lightest-blue",
        "id": "search-robots-button",
        "onChange": [Function],
        "placeholder": "search robots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          aria-label="Search"
          className="pa3 ba b--green bg-lightest-blue"
          id="search-robots-button"
          onChange={[Function]}
          placeholder="search robots"
          type="search"
/>,
        "className": "pa2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "Search",
          "className": "pa3 ba b--green bg-lightest-blue",
          "id": "search-robots-button",
          "onChange": [Function],
          "placeholder": "search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenTwoAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
